cmd/protoc-gen-connect-go/main.go

```diff
@@ -150,7 +150,9 @@ func generatePreamble(g *protogen.GeneratedFile, file *protogen.File) {
 	leadingComments(g, protogen.Comments(syntaxLocation.LeadingComments), false /* deprecated */)
 	g.P()
 
-	g.P("// Code generated by ", filepath.Base(os.Args[0]), ". DO NOT EDIT.")
+	programName := filepath.Base(os.Args[0])
+	programNameWithoutExt := strings.TrimSuffix(programName, filepath.Ext(programName))
+	g.P("// Code generated by ", programNameWithoutExt, ". DO NOT EDIT.")
```

Comment:
If the file is renamed to have a dot on a non-Windows system, this will remove everything after the dot. I think maybe we should be more targeted (and also maybe add a comment about why this trim call is present):
```suggestion
	// Remove .exe suffix on Windows so that generated code is stable, regardless
	// of whether it was generated on a Windows machine or not.
	if ext := filepath.Ext(programName); strings.ToLower(ext) == ".exe" {
		programName = strings.TrimSuffix(programName, ext)
	}
	g.P("// Code generated by ", programName, ". DO NOT EDIT.")
```